modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -844,6 +844,107 @@ def test_reset_index(data):
+            [[111.111, 222.222], ["col_one", "col_two"], [5, 6]],
+            names=["float", "string", "int"],
+        )
+        modin_df = pd.DataFrame(simple_data, index=index, columns=columns)
+        pandas_df = pandas.DataFrame(simple_data, index=index, columns=columns)
+    else:
+        modin_df = pd.DataFrame(data)
+        pandas_df = pandas.DataFrame(data)
+
+    col0 = modin_df.columns[0]
+    col1 = modin_df.columns[1]
+    pdf = pandas_df.groupby([col0, col1]).count()
+    pandas_reset = pdf.reset_index()
+    mdf = modin_df.groupby([col0, col1]).count()
+    modin_reset = mdf.reset_index()
+    df_equals(modin_reset, pandas_reset)
+
+    if data != "simple":
+
+        def make_index_from_3_columns(df, index_prefix):
+            # NaNs in column names lead to all kinds of weird behavior unrelated to this test
+            df.fillna(value=0, inplace=True)
+            # Filter out duplicated occurrences in first 3 columns
+            df = df.loc[~df.iloc[:, 0:3].duplicated()]
+            # Set index to use 3-levels from first 3 columns contents
+            df.set_index([df.columns[0], df.columns[1], df.columns[2]], inplace=True)
+            # Assign new multiindex level names
+            df.index.set_names(
+                [index_prefix + "_1", index_prefix + "_2", index_prefix + "_3"],
+                inplace=True,
+            )
+            return df
+
+        modin_df = make_index_from_3_columns(modin_df, "columns")
+        pandas_df = make_index_from_3_columns(pandas_df, "columns")
+        df_equals(modin_df, pandas_df)
+        modin_df = modin_df.T
+        pandas_df = pandas_df.T
+        df_equals(modin_df, pandas_df)
+        modin_df = make_index_from_3_columns(modin_df, "index")
+        pandas_df = make_index_from_3_columns(pandas_df, "index")
+        df_equals(modin_df, pandas_df)
+
+    kwargs = {}
+    if level != "no_level":
+        kwargs["level"] = level
+    if col_level != "no_col_level":
+        kwargs["col_level"] = col_level
+    if col_fill != "no_col_fill":
+        kwargs["col_fill"] = col_fill
+    pandas_reset = pandas_df.reset_index(**kwargs)
+    modin_reset = modin_df.reset_index(**kwargs)
+    df_equals(modin_reset, pandas_reset)
+
+
+def test_reset_index_with_named_index():
+    modin_df = pd.DataFrame(test_data_values[0])
+    pandas_df = pandas.DataFrame(test_data_values[0])
+
+    modin_df.index.name = pandas_df.index.name = "NAME_OF_INDEX"
+    df_equals(modin_df, pandas_df)
+    df_equals(modin_df.reset_index(drop=False), pandas_df.reset_index(drop=False))
+
+    modin_df.reset_index(drop=True, inplace=True)
+    pandas_df.reset_index(drop=True, inplace=True)
+    df_equals(modin_df, pandas_df)
+
+    modin_df = pd.DataFrame(test_data_values[0])
+    pandas_df = pandas.DataFrame(test_data_values[0])
+    modin_df.index.name = pandas_df.index.name = "NEW_NAME"
+    df_equals(modin_df.reset_index(drop=False), pandas_df.reset_index(drop=False))
+
+
```

Comment:
```suggestion
@pytest.mark.parametrize(
    "level",
    [None, 0, 1, 2, [2, 0], [1, 1]],
)
@pytest.mark.parametrize("col_level", [0, 1, 2])
@pytest.mark.parametrize("col_fill", ["", "name", 5, 5.5, None])
def test_reset_index_with_multi_index_no_drop(level, col_level, col_fill):
    data = test_data_values[0]
    index = generate_multiindex(len(data[list(data.keys())[0]]), nlevels=3)
    columns = generate_multiindex(len(data.keys()), nlevels=3)

    md_df, pd_df = create_test_dfs(data, index=index)
    eval_general(
        *create_test_dfs(data, index=index, columns=columns),
        lambda df: df.reset_index(
            level=level, drop=False, col_level=col_level, col_fill=col_fill
        ),
    )


@pytest.mark.parametrize(
    "level",
    [None, 0, 1, 2, [2, 0], [1, 1]],
)
def test_reset_index_with_multi_index_drop(level):
    data = test_data_values[0]
    index = generate_multiindex(len(data[list(data.keys())[0]]), nlevels=3)

    md_df, pd_df = create_test_dfs(data, index=index)
    eval_general(
        *create_test_dfs(data, index=index),
        lambda df: df.reset_index(level=level, drop=True),
    )


@pytest.mark.parametrize("is_multiindex", [True, False])
@pytest.mark.parametrize("is_name_multiindex", [True, False])
def test_reset_index_with_named_index(is_multiindex, is_name_multiindex):
    data = test_data_values[0]
    index = (
        generate_multiindex(len(data[list(data.keys())[0]]), nlevels=3)
        if is_multiindex
        else pandas.Index(np.arange(len(data[list(data.keys())[0]]), 0, -1))
    )
    columns = generate_multiindex(len(data.keys()), nlevels=3)
    index_names = (
        [(f"level{i}", 0) for i in range(index.nlevels)]
        if is_name_multiindex
        else [f"level{i}" for i in range(index.nlevels)]
    )

    index.names = index_names
    eval_general(
        *create_test_dfs(data, index=index, columns=columns),
        lambda df: df.reset_index(drop=False),
    )


```
previous version of testing contains more than 900 and didn't cover all cases, tests from suggestion contains about ~100 tests and revealed many of failing cases:
```
pytest test_indexing.py -k "test_reset_index"
========= 37 failed, 65 passed, 118 deselected, 105 warnings in 40.44s =========
```

Reply:
Ok I agree that some multiindex configurations were missing in the test, but I don't agree that with dropping a lot of parameter combinations. The fast that they pass the test doesn't mean that they aren't there for a reason. I would rather add a case with multilevel column names (unequal to number of column levels in the frame, BTW it also makes sense to add a case with more levels in multiindex than in the frame, not just a case with less levels) to my test than switch to your proposal.